dcos_installer/action_lib.py

```diff
@@ -371,38 +371,129 @@ def uninstall_dcos(config, block=False, state_json_dir=None, async_delegate=None
 
 
 def _add_prereqs_script(chain):
-    inline_script = """
-#/bin/sh
-# setenforce is in this path
-PATH=$PATH:/sbin
+    inline_script = r"""
+#!/usr/bin/env bash
 
-dist=$(cat /etc/os-release | sed -n 's@^ID="\(.*\)"$@\\1@p')
+# Exit on error, unset variable, or error in pipe chain
+set -o errexit -o nounset -o pipefail
 
-if ([ x$dist == 'xcoreos' ]); then
-  echo "Detected CoreOS. All prerequisites already installed" >&2
+# For setenforce & xfs_info
+PATH=$PATH:/usr/sbin:/sbin
+
+if [[ -f /opt/dcos-prereqs.installed ]]; then
+  echo "install_prereqs has been already executed on this host, exiting..."
   exit 0
 fi
 
-if ([ x$dist != 'xrhel' ] && [ x$dist != 'xcentos' ]); then
-  echo "$dist is not supported. Only RHEL and CentOS are supported" >&2
+echo "Validating distro..."
+distro="$(cat /etc/os-release | sed -n 's/^ID="\(.*\)"$/\1/p')"
```

Comment:
@karlkfi I am running the CoreOS `alpha` build, and the above `sed` doesn't seem to work:
```
dist=$(cat /etc/os-release | sed -n 's@^ID="\(.*\)"$@\\1@p')
```
These are the content of the `/etc/os-release`:
```
NAME="Container Linux by CoreOS"
ID=coreos
VERSION=1562.1.0
VERSION_ID=1562.1.0
BUILD_ID=2017-10-12-0536
PRETTY_NAME="Container Linux by CoreOS 1562.1.0 (Ladybug)"
ANSI_COLOR="38;5;75"
HOME_URL="https://coreos.com/"
BUG_REPORT_URL="https://issues.coreos.com"
COREOS_BOARD="amd64-usr"
```
However, when I do:
```
 cat /etc/os-release | sed -n 's@^ID=\(.*\)@\1@p'
coreos
```
It works fine. The quotes shouldn't be present in the `sed`?

Reply:
Sounds like the quotes need to be optional in the sed, because CentOS definitely has them. I guess these are env vars. So if there's no space in the string, the quotes are optional.

Reply:
Since these are env vars, how about we skip the `sed` and just source the file? One less regex to think about.

```
dist=$(. /etc/os-release && echo $ID)
```

Reply:
We might be better off with `distro="$(source /etc/os-release && echo $ID)"`

Reply:
done. I also changed the other one later on since the VERSION_ID is always in the same file on the 3 OSs we support.